rotkehlchen/db/snapshots.py

```diff
@@ -223,27 +226,33 @@ def update(
             location_data_snapshot: List[LocationData],
     ) -> Tuple[bool, str]:
         """Updates a DB Balance snapshot at a given timestamp."""
-        # delete the existing snapshot of that timestamp
-        is_success, message = self.delete(timestamp)
-        if is_success is False:
-            return is_success, message
+        with self.db.user_write() as cursor:
+            # delete the existing snapshot of that timestamp
+            is_success, message = self.delete(
+                write_cursor=cursor,
+                timestamp=timestamp,
+            )
+            if is_success is False:
+                return is_success, message
 
-        # update the snapshot with the provided data.
-        is_success, msg = self.import_snapshot(
-            processed_balances_list=balances_snapshot,
-            processed_location_data_list=location_data_snapshot,
-        )
+            # update the snapshot with the provided data.
+            is_success, msg = self.import_snapshot(
+                write_cursor=cursor,
+                processed_balances_list=balances_snapshot,
+                processed_location_data_list=location_data_snapshot,
+            )
+            if is_success is False:
+                self.db.conn.rollback()
```

Comment:
No this is the wrong approach. The function should be rewritten to make sure that exceptions are not caught in the fucntion but are propagated outside so the context manager of the cursor can do its job.